### PR TITLE
WT-11844 Sort string.ok and spell.ok

### PR DESCRIPTION
**Pull request by etienneptl**
Please see the JIRA ticket for the description of the annoyance faced without those changes.

### CODE REVIEW

File: dist/s_string [hunk 1]
```diff
@@ -67,20 +69,20 @@ while :
         # Between aspell versions 0.60.7-20110707 and 0.60.8 the following program names have been
         # added to the dictionary and are no longer considered spelling errors. This means
         # that building a replacement list of words with the newer version of aspell will omit them
-        # from s_string.ok, and subsequently running s_string with older aspell version will
+        # from our custom words, and subsequently running s_string with older aspell version will
```
Comment (ajmorton) [inline dist/s_string:72]:
```suggestion
        # from our custom words file, and subsequently running s_string with older aspell version will
```

File: dist/s_docs [hunk 2]
```diff
@@ -87,7 +87,18 @@ structurechk()
 spellchk()
 {
     # If aspell has been installed, run a spell check.
-    type aspell > /dev/null 2>&1 || return
+    type aspell > /dev/null 2>&1 || exit 1
```
Comment (ajmorton) [inline dist/s_docs:90]:
```suggestion
    type aspell > /dev/null 2>&1 || (echo "Cannot find aspell!"; exit 1)
```

File: dist/s_docs [hunk 3]
```diff
@@ -87,7 +87,18 @@ structurechk()
...
+
+    # Make sure custom spellings are sorted.
+    header="personal_ws-1.1 en 200"
+    custom_spellings=../src/docs/spell.ok
+
+    # We need to remove the required header first.
+    sed -i '/personal_ws-1.1 en 200/d' $custom_spellings
+    sort -c $custom_spellings || echo "Sorting $custom_spellings..." && sort -u -o $custom_spellings $custom_spellings
```
Comment (ajmorton) [inline dist/s_docs:98]: Small nit: I find the following more readable. Similarly in s_string
```suggestion
    if ! sort -c $custom_spellings; then
        echo "Sorting $custom_spellings..."
        sort -u -o $custom_spellings $custom_spellings
    fi
```

**Review by ajmorton — APPROVED** [review]
LGTM! Some suggestions but good to merge as is

File: dist/s_docs [hunk 4]
```diff
@@ -87,7 +87,21 @@ structurechk()
...
-    type aspell > /dev/null 2>&1 || return
+    type aspell > /dev/null 2>&1 || (echo "Cannot find aspell!"; exit 1)
+
+    # Make sure custom spellings are sorted.
+    header="personal_ws-1.1 en 200"
+    custom_spellings=../src/docs/spell.ok
+
+    # We need to remove the required header first.
+    sed -i '/personal_ws-1.1 en 200/d' $custom_spellings
+    if ! sort -c $custom_spellings; then
```
Comment (ershov) [inline dist/s_docs:98]: I know that `aspell` doesn't work with language `C` but everything else relies on `C` sorting order. So let's set `LC_ALL=C` for `sort` only: `LC_ALL=C sort -c $custom_spellings`.

Or alternatively, we can set globally `LC_ALL=C` in the file but call aspell with `LC_ALL=EN_us aspell ...`.

**Review by ershov — APPROVED** [review]

**Comment by etienneptl** [comment]
evergreen merge

**Pull request merged**